flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigManager.java

```diff
@@ -133,6 +135,19 @@ public Configuration getObserveConfig(FlinkDeployment deployment) {
         return getConfig(deployment.getMetadata(), ReconciliationUtils.getDeployedSpec(deployment));
     }
 
+    public Configuration getSessionJobObserveConfig(
+            FlinkDeployment deployment, FlinkSessionJob flinkSessionJob) {
+        Configuration sessionJobConfig = getObserveConfig(deployment);
+
+        // merge session job specific config
+        Map<String, String> sessionJobFlinkConfiguration =
+                flinkSessionJob.getSpec().getFlinkConfiguration();
+        if (sessionJobFlinkConfiguration != null && !sessionJobFlinkConfiguration.isEmpty()) {
+            sessionJobFlinkConfiguration.forEach(sessionJobConfig::setString);
```

Comment:
I am wondering for this particular case, would it make sense to merge the HTTP headers instead of overwriting them if the base config also defined it?

Reply:
I guess the logic would be more straight forward if this config in sessionJob directly overrides the parent config. 

If we do the merging work. Then a user will need to check two CRs instead of one to determine what are the actual headers that is applied during jar fetching during debugging. It could a little bit confusing.